### ISSUE TITLE
Implicit behavior when parsing malformed rubocop:disable directives

### ISSUE THREAD

**Issue by sksmsWKd** [rubocop/rubocop#14794]
### Summary

While checking how `rubocop:disable` / `rubocop:enable` directives are parsed, i noticed some implicit behavior when the `disable` directive is malformed or loosely formatted.

I am raising this as a question and a concern, rather than as a report of incorrect behavior, to better understand the intended RuboCop's design, behavior and culture.

### Observed behavior

The following examples are all accepted and processed without warnings:

```ruby
# rubocop:disable Layout/LineLength:foo
# rubocop:enable Layout/LineLength

# rubocop:disable Layout/LineLength bar
# rubocop:enable Layout/LineLength

# rubocop:disable Layout/LineLength Metrics/AbcSize
# rubocop:enable Layout/LineLength

# rubocop:disable Layout/LineLength Metrics/AbcSize, Metrics/CyclomaticComplexity
# rubocop:enable Layout/LineLength
```

In all cases above, only the first cop (Layout/LineLength) is actually disabled.
Any additional tokens or cop names in the disable directive appear to be ignored implicitly. (Without any warnings)

From a reader’s perspective, it can be difficult to understand which parts of a directive are meaningful when malformed input is accepted silently.

Because additional tokens (strings) are ignored without feedback, it is not always obvious whether a directive was written as intended or partially ignored.

This is not necessarily incorrect behavior, but it does rely on implicit parsing rules that are not obvious from reading the code or documentation: these directives affect cop state at the file level, implicit handling of malformed input may have effects beyond the immediately surrounding lines.

### Questions & Discussions

Is this implicit handling of malformed `rubocop:disable` directives intentional?

If so, would documenting this behavior help avoid confusion?

If not, is there a non-breaking way to make this behavior more explicit, while avoiding disruptive changes for existing RuboCop users?

### RuboCop version

`1.50.2 (using Parser 3.3.7.1, rubocop-ast 1.30.0, running on ruby 2.6.6) [-darwin22]`
